packages/node_modules/pouchdb-adapter-utils/src/updateDoc.js

```diff
@@ -32,7 +32,8 @@ function updateDoc(revLimit, prev, docInfo, results,
 
   var merged = merge(prev.rev_tree, docInfo.metadata.rev_tree[0], revLimit);
 
-  var inConflict = newEdits && (((previouslyDeleted && deleted) ||
+  var inConflict = newEdits && ((
+    (previouslyDeleted && deleted && merged.conflicts !== 'new_leaf') ||
     (!previouslyDeleted && merged.conflicts !== 'new_leaf') ||
     (previouslyDeleted && !deleted && merged.conflicts === 'new_branch')));
```

Comment:
This logic is starting to get really hairy, and I'm not sure it covers all cases. But on the bright side, it covers everything in our unit tests.